docs/api-reference/carto/carto-layer.md

```diff
@@ -153,6 +163,32 @@ Receives arguments:
 
 - `error` (`Error`)
 
+### Sub Layers
+
+The `CartoLayer` is a [`CompositeLayer`](/docs/api-reference/core/composite-layer.md), and will generate different sublayers depending on the configuration. In all cases, properties of the [`GeoJsonLayer`](/docs/api-reference/layers/geojson-layer.md) will be inherited.
+
+#### API v1 & v2
+
+`CartoLayer` works with the different CARTO Maps API versions (v1, v2, and v3). When using version v1 and v2, the layer always works with vector tiles so it inherits all properties from [`MVTLayer`](/docs/api-reference/geo-layers/mvt-layer.md).
+
+#### API v3
+
+When using v3, the behavior depends on the `type` property.
+
+##### `MAP_TYPES.QUERY`
+
+GeoJSON/JSON data will be used, depending on `format`. No additional properties inherited.
+
+##### `MAP_TYPES.TILESET`
+
+Tiled data will be used, depending on `formatTiles`. All [`MVTLayer`](/docs/api-reference/geo-layers/mvt-layer.md) properties will be inherited.
+
+##### `MAP_TYPES.TABLE`
+
+If `format` is `FORMATS.TILEJSON`, or the table is large, dynamic tiling is enabled and the behavior is as per `MAP_TYPES.TILESET`. Otherwise behavior is as per `MAP_TYPES.QUERY`.
+
+the layer works with vector tiles if the `type` property is `MAP_TYPES.TILESET` and with GeoJSON data if the `type` is `MAP_TYPES.QUERY`. If the type is `MAP_TYPES.TABLE`.
```

Comment:
Remove this line

Reply:
I think referring to QUERY or TILESET is difficult to understand. I'd change to something like this:

Automatically based on the size of the table we decide if use tiles or a full document.

If we use a full document we use the GeoJSONLayer, if we use tiles we use a vector tile layer that extends the current MVTLayer and inherits all the properties.

You can force to use tiles by setting format to FORMATS.TILEJSON

Reply:
Looks great